06_puppy/alextmz/mapstore.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"errors"
+)
+
+type MapStore map[uint16]*Puppy
+
+func NewmapStore() *MapStore {
+	a := MapStore{}
+	return &a
+}
+
+func (m *MapStore) CreatePuppy(p *Puppy) error {
+	if _, ok := (*m)[p.ID]; !ok {
```

Comment:
@alextmz keen to understand why you choose to access the puppy ID as (*m)[p.ID] instead of using p.ID?

Reply:
Some of the CO's pointed out in my review that a good design choice would be to get the API handle the puppy id creation instead of the user providing the ID.

Reply:
@sirigithub Not sure if I understand your question - would not make sense to use p.ID as it is the item received, whereas (*m) is the Mapstore we're adding the item to. This specific statement checks if our mapstore already has a item with the same ID as the item received.
Indeed the API handling the ID would be a better approach. I'll look into it, thanks!

Reply:
@alextmz oops my bad i mis read the syntax, pl ignore my first comment.